spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -799,6 +799,9 @@ def write_data_spec(
 
         self._weight_scales[placement] = weight_scales
 
+    def clear_connection_cache(self):
+        self._retrieved_blocks = dict()
```

Comment:
hes impling that this change doesnt fix the get weights overall. i cant see how it would fail given its the blunt force right now. and testing a static syanpse is being reread when the data is identical in master....... im more inclined to believe static strucal is bust right now, but peters on the case to check